dmoj/cptbox/ptdebug_arm.cpp

```diff
@@ -26,7 +26,11 @@ int pt_debugger_arm::syscall() {
 }
 
 void pt_debugger_arm::syscall(int id) {
+#ifdef PTRACE_SET_SYSCALL
+    ptrace(PTRACE_SET_SYSCALL, tid, 0, id);
+#else
     poke_reg(ARM_r7, id);
```

Comment:
"¬`PTRACE_SET_SYSCALL` ⇒ compiling on ARM" is a pretty convoluted assumption, I'd reverse it to `#ifdef PTBOX_ARM` or something like that instead.

Reply:
Our current setup requires compiling ARM stuff on x86, so we need some silly hacks.